### DISCUSSION TITLE
PostCSS 8 support

### DISCUSSION THREAD

**Discussion by JoshuaCS94** [storybookjs/storybook#13971] [Help] [answered]
Hi, guys! :wave: I just started a new project from scratch. I would like to develop it on Next and Tailwind and use Storybook for developing and testing my components. I created the Next v10 project with `npx create-next-app my-project` and installed Tailwind v2, like the [official guide](https://tailwindcss.com/docs/guides/nextjs) says, including PostCSS 8 and autoprefixer, and everything seems to work fine. The problem is I can't or don't know how to setup Storybook. I tried with the regular `npx sb init` but, whenever I do `npm run storybook`, at the end of the build process, it throws a bunch of errors but I can spot one line regarding PostCSS 8:

```
ERROR in ./node_modules/tailwindcss/tailwind.css (./node_modules/css-loader/dist/cjs.js??ref--11-1!./node_modules/@storybook/core/node_modules/postcss-loader/src??ref--11-2!./node_modules/tailwindcss/tailwind.css)
Module build failed (from ./node_modules/@storybook/core/node_modules/postcss-loader/src/index.js):
Error: PostCSS plugin tailwindcss requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
    at Processor.normalize (/joshua/0. Ciencia de la computacion/- Sibucán/web-app/node_modules/@storybook/core/node_modules/postcss/lib/processor.js:153:15)
    at new Processor (/joshua/0. Ciencia de la computacion/- Sibucán/web-app/node_modules/@storybook/core/node_modules/postcss/lib/processor.js:56:25)
    at postcss (/joshua/0. Ciencia de la computacion/- Sibucán/web-app/node_modules/@storybook/core/node_modules/postcss/lib/postcss.js:55:10)
    at /joshua/0. Ciencia de la computacion/- Sibucán/web-app/node_modules/@storybook/core/node_modules/postcss-loader/src/index.js:140:12
 @ ./node_modules/tailwindcss/tailwind.css 2:26-161 53:4-74:5 56:18-153
 @ ./.storybook/preview.js
 @ ./.storybook/preview.js-generated-config-entry.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/storybook-init-framework-entry.js ./node_modules/@storybook/addon-docs/dist/frameworks/common/config.js-generated-other-entry.js ./node_modules/@storybook/addon-docs/dist/frameworks/react/config.js-generated-other-entry.js ./node_modules/@storybook/addon-links/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/preset/addArgs.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/preset/addParameter.js-generated-other-entry.js ./.storybook/preview.js-generated-config-entry.js ./.storybook/generated-stories-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined
```

So, my question is, does Storybook support PostCSS 8 or not? What do I have to do in order to get this all up and running?

If you need further information, just ask for it, like showing my config files. Thanks in advance!

**Comment by shilman** [comment, marked as answer]
Please read the instructions here https://storybook.js.org/addons/@storybook/addon-postcss

**Reply by wagerfield** [reply]
@shilman I followed the steps in this documentation and am still having issues relating to `postcss-loader` I believe.

I have created a bare bones reproduction repo with Next.js, Storybook, Tailwind 2 and PostCSS 8 with the `@storybook/addon-postcss` addon configured exactly as documented and dropped my notes and observations in the readme:

https://github.com/wagerfield/storybook-tailwind

If you wouldn't mind taking a look at this, I would be eternally grateful 🙏

**Reply by shilman** [reply]
@wagerfield the problem is that you are using the wrong version of storybook. the PostCSS 8 fixes are in 6.2, so i changed the version specifiers and it works:

```
    "@storybook/addon-essentials": "next",
    "@storybook/react": "next",
```

we'll be releasing a stable 6.2.0 in ~2 weeks

**Reply by wagerfield** [reply]
You fine Sir are a legend! Thanks so much for your help 🙏 

I'll update the readme and deps to a working solution for anyone else looking to get this working.

I can also confirm that the newly released `@tailwindcss/jit` compiler works with this configuration which is beyond awesome.

Have a stellar day 🚀

**Comment by marcGiroux83** [comment]
still does not work with 6.3.6. still getting

Error: PostCSS plugin tailwindcss requires PostCSS 8.

**Reply by namvhoang02** [reply]
Same here. @shilman Could you please take a look at this?

**Reply by shilman** [reply]
Sure, where's a reproduction I can look at?

**Reply by marcGiroux83** [reply]
doing the same thing as @wagerfield  but using the latest version of @storybook/addon-postcss and the version is still not the right one. so unless something i'm missing?

this is from the master branch on the postcss addon git (package.json)

![Screen Shot 2021-08-02 at 9 23 47 AM](https://user-images.githubusercontent.com/33124675/127868867-639f9aa7-f4bb-4a8a-8d5f-50e3ca4eae13.png)

**Reply by namvhoang02** [reply]
hi guys,
the issue has gone after I add @storybook/builder-webpack5 and webpack in my package.
 I noticed that @storybook/builder-webpack4 still use postcss@^7.x.x
Here is my commands:

```
npx sb init -f --builder webpack5
yarn add -D webpack
```

**Reply by Taelkir** [reply]
> hi guys,
> the issue has gone after I add @storybook/builder-webpack5 and webpack in my package.
>  I noticed that @storybook/builder-webpack4 still use postcss@^7.x.x
> Here is my commands:
> 
> ```
> npx sb init -f --builder webpack5
> yarn add -D webpack
> ```

No luck for me with this one unfortunately

**Reply by tyirenkyi** [reply]
> hi guys, the issue has gone after I add @storybook/builder-webpack5 and webpack in my package.  I noticed that @storybook/builder-webpack4 still use postcss@^7.x.x Here is my commands:
> 
> ```
> npx sb init -f --builder webpack5
> yarn add -D webpack
> ```

this worked for me. `"storybook-react":"^6.5.9"`

**Reply by gmonte** [reply]
I made it work just by adding this in my **package.json**:

```json
{
  "devDependencies": {
    "postcss": "^8.4.19"
  },
  "resolutions": {
    "postcss": "^8.4.19"
  }
}
```
